docs/src/main/asciidoc/security-oidc-code-flow-authentication-concept.adoc

```diff
@@ -381,12 +381,14 @@ quarkus.oidc.client-id=quarkus-app
 quarkus.oidc.credentials.secret=secret
 quarkus.oidc.application-type=web-app
 quarkus.oidc.token-state-manager.split-tokens=true
-quarkus.oidc.token-state-manager.encryption-required=true
 quarkus.oidc.token-state-manager.encryption-secret=eUk1p7UB3nFiXZGUXi0uph1Y9p34YhBU
 ----
 
-The token encryption secret must be 32 characters long. Note that you only have to set `quarkus.oidc.token-state-manager.encryption-secret` if you prefer not to use
-`quarkus.oidc.credentials.secret` for encrypting the tokens or if `quarkus.oidc.credentials.secret` length is less than 32 characters.
+The token encryption secret must be at least 32 characters long. If this key is not configured then either `quarkus.oidc.credentials.secret` or `quarkus.oidc.credentials.jwt.secret` will be hashed to create an encryption key.
+
+`quarkus.oidc.token-state-manager.encryption-secret` should be configured if Quarkus authenticates to OpenId Connect Provider using either mTLS or `private_key_jwt` (where a private RSA or EC key is used to sign a JWT token) authentication methods, otherwise a random key will be generated which will be problematic if the Quarkus application is running in the cloud with multiple pods managing the requests.
```

Comment:
I'm not sure if forcing encryption is the best thing to do from a UX perspective. I definitely agree it is better to avoid leaking sensitive data in cookies but perhaps for the majority of deployments (e.g.: enterprise use cases) where requests come from a trusted network, that is not a big issue.

Reply:
Hi @pedroigor, thanks for the feedback, right, I share some of your concerns,

>I'm not sure if forcing encryption is the best thing to do from a UX perspective. 

I've tried to ensure here that Quarkus users simply won't notice the encryption is happening by default now if they are not specifically configuring the encryption key, you can check a few modified tests, where the tests had to be fixed to decrypt the session cookie (for the test itself doing some session cookie checks to pass) without the encryption key directly configured. As noted above, if it is not configured - the fallback goes to the client secret, which is not guaranteed to be 32 chars long, but the encryption will still be hard to break and like you imply, we do expect, in production, that for ex either a trusted network is used, or at the very least, HTTPS, which cancels all/most of the risks anyway.
And if neither encryption key nor client secret is configured, then we are really talking about MTLS or private_key_jwt authentication between Quarkus and Keycloak or other OIDC provider - in this case the key will be auto-generated - this is the only/minor migration concern in a very high throughput, multi-pod setup where the pod which tries to decrypt the session cookie is not the one which encrypted it - but the probability of it happening is quite low, and in these cases users might want to disable the encryption or configure the encryption key, etc

>My point is, why not keep encryption optional but highly recommended in docs?

We are getting some push-back related to Quarkus Secure By Default strategy not proactively enforced

Reply:
Works for me too. Just wanted to highlight this concern. It is a trade-off and being more strict is good.